test/integration_test.go

```diff
@@ -512,6 +518,32 @@ func (ts *IntegrationTestSuite) TestDomainUpdate() {
 	ts.Equal(description, *domain.DomainInfo.Description)
 }
 
+func (ts *IntegrationTestSuite) TestNonDeterministicWorkflowFailPolicy() {
+	err := ts.executeWorkflow("test-nondeterminism-failpolicy", ts.workflows.NonDeterminismSimulatorWorkflow, nil)
+	var customErr *internal.CustomError
+	ok := errors.As(err, &customErr)
+	ts.Truef(ok, "expected CustomError but got %T", err)
+	ts.Equal("NonDeterministicWorkflowPolicyFailWorkflow", customErr.Reason())
+}
+
+func (ts *IntegrationTestSuite) TestNonDeterministicWorkflowQuery() {
+	ctx, cancel := context.WithTimeout(context.Background(), ctxTimeout)
+	defer cancel()
+	run, err := ts.libClient.ExecuteWorkflow(ctx, ts.startWorkflowOptions("test-nondeterministic-query"), ts.workflows.NonDeterminismSimulatorWorkflow)
+	ts.Nil(err)
+	err = run.Get(ctx, nil)
+	customErr, ok := err.(*internal.CustomError)
+	ts.Truef(ok, "expected CustomError but got %T", err)
+	ts.Equal("NonDeterministicWorkflowPolicyFailWorkflow", customErr.Reason())
+
+	// query failed workflow should still work
+	value, err := ts.libClient.QueryWorkflow(ctx, "test-nondeterministic-query", run.GetRunID(), "__stack_trace")
+	ts.NoError(err)
+	ts.NotNil(value)
+	var trace string
+	ts.Nil(value.Get(&trace))
```

Comment:
```suggestion
	ts.NoError(value.Get(&trace))
```
I assume that's the intent here?  might as well be a bit closer to type-correctness / it's relevant that it's not an error, not just an [anything not nil].

you can also make this not panic in the nil case with a `require.NotNil(s.T(), value)` earlier.  not particularly important since it's a fairly small error and the stack will point out the issue, just as an FYI.

Reply:
following the same pattern where `QueryWorkflow` is used but will switch to `ts.NoError`